internal/rbd/replicationcontrollerserver.go

```diff
@@ -104,6 +119,43 @@ func getMirroringMode(ctx context.Context, parameters map[string]string) (librbd
 	return mirroringMode, nil
 }
 
+// getSchedulingDetails gets the mirroring mode and scheduling details from the
+// input GRPC request parameters and validates the scheduling is only supported
+// for mirroring mode.
+func getSchedulingDetails(parameters map[string]string) (admin.Interval, admin.StartTime, error) {
+	val := parameters[imageMirroringKey]
+
+	switch imageMirroringMode(val) {
+	case imageMirrorModeSnapshot:
+	default:
+		return "", "", status.Error(codes.InvalidArgument, "scheduling is only supported for snapshot mode")
+	}
+
+	startTime := parameters[schedulingStartTimeKey]
+	// validate mandatory interval field
+	interval, ok := parameters[schedulingIntervalKey]
+	if ok && interval == "" {
+		return "", "", status.Error(codes.InvalidArgument, "scheduling interval cannot be empty")
+	}
+	if !ok {
+		// startTime is alone not supported it has to be present with interval
+		if startTime != "" {
+			return "", "", status.Errorf(codes.InvalidArgument, "%s parameter is supported only with %s", schedulingStartTimeKey, schedulingIntervalKey)
```

Comment:
nit: instead of `%s`, using `%q` might be nicer. Also, this line looks pretty long :thinking: